src/util/config.js

```diff
@@ -1,83 +1,157 @@
-			'https://secure.dev.meetup.com/oauth2/access',
-		PHOTO_SCALER_SALT: process.env.PHOTO_SCALER_SALT,
-		oauth: {
-			secret: process.env.MUPWEB_OAUTH_SECRET,
-			key: process.env.MUPWEB_OAUTH_KEY,
+const oauthError = new Error('get oauth secrets from #web-platform team');
+
+let config = convict({
+	env: {
+		format: ['production', 'development', 'test'],
+		default: 'development',
+		env: 'NODE_ENV'
+	},
+	api: {
+		protocol: {
+			format: function(protocol) {
+				if (!['http', 'https'].includes(protocol)) {
+					throw new Error('must be http or https');
+				}
+			},
+			default: 'https',
+			env: 'API_PROTOCOL',
+		},
+		host: {
+			format: String,
+			default: 'api.dev.meetup.com',
+			env: 'API_HOST'
+		},
+		timeout: {
+			format: 'int',
+			default: 8000,
+			env: 'API_TIMEOUT'
+		}
+	},
+	asset_server: {
+		host: {
+			format: String,
+			default: '0.0.0.0',
+			env: 'ASSET_SERVER_HOST'
+		},
+		port: {
+			format: 'port',
+			default: 8001,
+			env: 'ASSET_SERVER_PORT'
+		},
+	},
+	cookie_encrypt_secret: {
+		format: function (secret) {
+			if (secret.toString().length < 32) {
+				throw new Error('set COOKIE_ENCRYPT_SECRET env variable to a random 32+ character string');
+			}
+		},
+		default: null,
+		env: 'COOKIE_ENCRYPT_SECRET'
+	},
+	csrf_secret: {
+		format: function (secret) {
+			if (secret.toString().length < 32) {
+				throw new Error('set CSRF_SECRET env variable to a random 32+ character string');
+			}
+		},
+		default: null,
+		env: 'CSRF_SECRET'
+	},
+	dev_server: {
+		host: {
+			format: String,
+			default: '0.0.0.0',
+			env: 'DEV_SERVER_HOST'
 		},
-	};
-	config.duotoneUrls = getDuotoneUrls(duotones, config.PHOTO_SCALER_SALT);
-	config.API_SERVER_ROOT_URL = `${config.API_PROTOCOL}://${config.API_HOST}`;
+		port: {
+			format: 'port',
+			default: 8000,
+			env: 'DEV_SERVER_PORT'
+		},
+	},
+	oauth: {
+		auth_url: {
+			format: 'url',
+			default: 'https://secure.dev.meetup.com/oauth2/authorize',
+			env: 'OAUTH_AUTH_URL'
+		},
+		access_url: {
+			format: 'url',
+			default: 'https://secure.dev.meetup.com/oauth2/access',
+			env: 'OAUTH_ACCESS_URL'
+		},
+		secret: {
+			format: function (secret) {
+				if (secret.toString().length < 1) {
+					throw oauthError;
+				}
+			},
+			default: null,
+			env: 'MUPWEB_OAUTH_SECRET'
+		},
+		key: {
+			format: function (key) {
+				if (key.toString().length < 1) {
+					throw oauthError;
+				}
+			},
+			default: null,
+			env: 'MUPWEB_OAUTH_KEY'
+		}
+	},
+	photo_scaler_salt: {
+		format: function (salt) {
+			if (salt.toString().length < 1) {
+				throw new Error('get PHOTO_SCALER_SALT from #web-platform team');
+			}
+		},
+		default: null,
+		env: 'PHOTO_SCALER_SALT',
+	}
+});
 
-	// currently all config is available syncronously, so resolve immediately
-	return Promise.resolve({ ...config, ...overrideConfig })
-		.then(validateConfig);
+// Load environment dependent configuration
+const env = config.get('env');
+const configFile = `../config.${env}.json`;
```

Comment:
Having a hard-coded 'default' config file location is probably best to start - I like the idea of passing in a config file path as a command line arg, but let's aim to do interface optimization in the future.

_Who configs the config?_